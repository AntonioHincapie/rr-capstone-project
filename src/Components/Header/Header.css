.Header {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1rem;
  background-color: var(--pink);
  color: var(--white);
}

.Header h3 {
  font-family: var(--lato);
  font-weight: bolder;
  font-size: 1.2rem;
}

.Header p {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  font-family: var(--gill);
  font-weight: 600;
  font-size: 0.9rem;
}

.Header p img {
  height: 25px;
  width: auto;
}
