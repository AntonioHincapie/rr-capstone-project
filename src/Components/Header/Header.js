import logo from '../imgs/coincap-logo.png';

export default function Header() {
  return (
    <div className="Header">
      <h3>CryptoCurrency Info</h3>
      <p>Built with <img src={logo} alt="coincap" />API and Cryptocurrency-Icons</p>
    </div>
  );
}
