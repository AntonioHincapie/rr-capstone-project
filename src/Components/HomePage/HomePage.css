.container {
  padding: 1rem;
}

.selector {
  display: flex;
  flex-direction: column;
  align-items: center;
}

#filter {
  padding: 0 4px;
  height: 2rem;
  width: 7rem;
  background-color: var(--pink);
  color: var(--white);
  border-radius: 15px;
  border: solid 2px var(--gray);
}

.cryptos {
  list-style: none;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
  margin: 1rem 0;
}

.cryptos li a {
  text-decoration: none;
}

.cryptoList {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1rem;
  border: solid 2px var(--gray);
  border-radius: 3px;
  padding: 1rem;
}

.cryptoList img {
  height: 32px;
  width: 32px;
}

.cryptoList h4 {
  font-family: var(--gill);
  font-weight: 500;
  font-size: 1rem;
  color: var(--white);
}
