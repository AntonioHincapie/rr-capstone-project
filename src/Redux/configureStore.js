import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import CrytoReducer from './GetCurrencyReducer';

const RootReducer = combineReducers({
  cryptos: CrytoReducer,
});

const store = createStore(RootReducer, applyMiddleware(thunk, logger));

export default store;
