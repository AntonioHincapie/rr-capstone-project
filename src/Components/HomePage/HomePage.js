import { useDispatch, useSelector } from 'react-redux';
import { useState } from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FetchCryptos } from '../../Redux/GetCurrencyReducer';
import './HomePage.css';

const Crypto = (props) => {
  const {
    id, img, name, rank,
  } = props;

  return (
    <li>
      <NavLink to={`/${id}`}>
        <div className="cryptoList">
          <img src={img} alt={name} />
          <h4>
            #
            {rank}
            {name}
          </h4>
        </div>
      </NavLink>
    </li>
  );
};

export default function HomePage() {
  const cryptos = useSelector((state) => state.cryptos);
  const dispatch = useDispatch();
  const [filterCrypto, setFilter] = useState([]);

  if (cryptos.length === 0) {
    dispatch(FetchCryptos());
  }

  const filterHandler = (e) => {
    if (e.target.value === '10') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 10));
    } else if (e.target.value === '20') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 20 && cryp.rank >= 11));
    } else if (e.target.value === '30') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 30 && cryp.rank >= 21));
    } else if (e.target.value === '40') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 40 && cryp.rank >= 31));
    } else if (e.target.value === '50') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 50 && cryp.rank >= 41));
    } else if (e.target.value === '60') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 60 && cryp.rank >= 51));
    } else if (e.target.value === '70') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 70 && cryp.rank >= 61));
    } else if (e.target.value === '80') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 80 && cryp.rank >= 71));
    } else if (e.target.value === '90') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 90 && cryp.rank >= 81));
    } else if (e.target.value === '100') {
      setFilter(cryptos.filter((cryp) => cryp.rank <= 100 && cryp.rank >= 91));
    } else {
      setFilter(cryptos);
    }
  };

  return (
    <div className="container">
      <div className="selector">
        <select name="filter" id="filter" onInput={(e) => filterHandler(e)}>
          <option defaultValue="All">All</option>
          <option value="10">1 to 10</option>
          <option value="20">11 to 20</option>
          <option value="30">21 to 30</option>
          <option value="40">31 to 40</option>
          <option value="50">41 to 50</option>
          <option value="60">51 to 60</option>
          <option value="70">61 to 70</option>
          <option value="80">71 to 80</option>
          <option value="90">81 to 90</option>
          <option value="100">91 to 100</option>
        </select>
      </div>
      <ul className="cryptos">
        {filterCrypto.length > 0
        && filterCrypto.map((crypto) => (
          <Crypto
            key={crypto.id}
            id={crypto.id}
            name={crypto.name}
            img={crypto.img}
            rank={crypto.rank}
          />
        ))}
        {filterCrypto.length === 0
        && cryptos.map((crypto) => (
          <Crypto
            key={crypto.id}
            id={crypto.id}
            name={crypto.name}
            img={crypto.img}
            rank={crypto.rank}
          />
        ))}
      </ul>
    </div>
  );
}

Crypto.propTypes = {
  id: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  rank: PropTypes.number.isRequired,
};
