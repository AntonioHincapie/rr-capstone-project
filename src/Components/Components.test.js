import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from '../Redux/configureStore';
import Header from './Header/Header';
import HomePage from './HomePage/HomePage';

describe('Components Test', () => {
  test('Header Test', () => {
    const store = configureStore;
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('HomePage Test', () => {
    const store = configureStore;
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
