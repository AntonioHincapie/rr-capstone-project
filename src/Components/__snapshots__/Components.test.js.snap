// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Test Header Test 1`] = `
<div
  className="Header"
>
  <h3>
    CryptoCurrency Info
  </h3>
  <p>
    Built with
    <img
      alt="coincap"
      src="coincap-logo.png"
    />
    API and Cryptocurrency-Icons
  </p>
</div>
`;

exports[`Components Test HomePage Test 1`] = `
<div
  className="container"
>
  <div
    className="selector"
  >
    <select
      id="filter"
      name="filter"
      onInput={[Function]}
    >
      <option
        defaultValue="All"
      >
        All
      </option>
      <option
        value="10"
      >
        1 to 10
      </option>
      <option
        value="20"
      >
        11 to 20
      </option>
      <option
        value="30"
      >
        21 to 30
      </option>
      <option
        value="40"
      >
        31 to 40
      </option>
      <option
        value="50"
      >
        41 to 50
      </option>
      <option
        value="60"
      >
        51 to 60
      </option>
      <option
        value="70"
      >
        61 to 70
      </option>
      <option
        value="80"
      >
        71 to 80
      </option>
      <option
        value="90"
      >
        81 to 90
      </option>
      <option
        value="100"
      >
        91 to 100
      </option>
    </select>
  </div>
  <ul
    className="cryptos"
  />
</div>
`;
